File: app/src/main/java/sugtao4423/lod/LoDBaseActivity.java
package sugtao4423.lod;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public abstract class LoDBaseActivity extends AppCompatActivity{

    protected App app;

    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        app = (App)getApplicationContext();
    }

    @Override
    protected void onResume(){
        super.onResume();
        app.getUseTime().start();
    }

    @Override
    protected void onPause(){
        super.onPause();
        app.getUseTime().stop();
    }

}

File: app/src/main/java/sugtao4423/lod/LoDBaseActivity.kt
package sugtao4423.lod

import androidx.appcompat.app.AppCompatActivity
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

abstract class LoDBaseActivity : AppCompatActivity() {

    private var useTimeStartTime = System.currentTimeMillis()

    override fun onResume() {
        super.onResume()
        useTimeStartTime = System.currentTimeMillis()
    }

    override fun onPause() {
        super.onPause()
        val useTimeInMillis = System.currentTimeMillis() - useTimeStartTime
        CoroutineScope(Dispatchers.Main).launch {
            (applicationContext as App).useTimeRepository.save(useTimeInMillis)
        }
    }
}
